lib/teleterm/clusters/storage.go

```diff
@@ -141,6 +141,7 @@ func (s *Storage) addCluster(ctx context.Context, dir, webProxyAddress string) (
 	cfg.WebProxyAddr = webProxyAddress
 	cfg.HomePath = s.Dir
 	cfg.KeysDir = s.Dir
+	cfg.ClientStore = client.NewFSClientStore(s.Dir)
```

Comment:
From what I see, `client.NewClient(cfg)` returns the configured `ClientStore`, so we could have
```typescript
cfg.ClientStore = clusterClient.ClientStore
``` 
Wouldn't it be better to use it?

Reply:
`clusterClient` doesn't exist at this point though. Ideally we wouldn't call `SaveProfile` on `Config` in the first place, I'm not sure why we're doing that – I left a comment to investigate this in the future.

Reply:
Ah, but I mean yeah, we could first create Config, then TeleportClient, then set the ClientStore on Config to that of TeleportClient.

It's a bit convoluted though IMHO. In case of ClientStore, the choice is between an in-memory implementation or an implementation that writes to disk. The whole reason we call `SaveProfile` is to save that to disk so that the subsequent calls to tshd see that cluster too. So I don't think it's bad that we're duplicating the logic a little bit here.

Reply:
Yeah, I realized that it would create some sort of a circlar dependency between these functions :/